File: server/models/Rehearsal.js
import mongoose from 'mongoose'
import Project from './Project'

const Schema = mongoose.Schema

const rehearsalSchema = new Schema({
    startDateTime: { type: 'Date', required: true },
    endDateTime: { type: 'Date', required: true },
    comment: { type: 'String' },
    isOpen: { type: 'Boolean' },
    isGeneral: { type: 'Boolean' },
    isDress: { type: 'Boolean' },
    isRehearsalDay: { type: 'Boolean' },
    isInConcertClothing: { type: 'Boolean' },
    withSoloist: { type: 'Boolean' },
    parts: [{ type: 'ObjectId', ref: 'RehearsalPart' }],
})

rehearsalSchema.pre('remove', next => {
    Project.update(
        { rehearsals: this._id },
        { $pull: { rehearsals: this._id } },
        { multi: true }
    ).exec()
    next()
})

export default mongoose.model('Rehearsal', rehearsalSchema, 'rehearsals')

File: src/main/kotlin/nl/krashna/app/models/Rehearsal.kt
package nl.krashna.app.models

import java.util.Date
import javax.persistence.CascadeType
import javax.persistence.Column
import javax.persistence.ElementCollection
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.Temporal
import javax.persistence.TemporalType

@Entity
data class Rehearsal(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long,

    @Column(nullable = false)
    @Temporal(TemporalType.TIMESTAMP)
    val startDateTime: Date,

    @Column(nullable = false)
    @Temporal(TemporalType.TIMESTAMP)
    val endDateTime: Date,

    @Column(nullable = false)
    val comment: String,

    @Column(nullable = false)
    val isOpen: Boolean,

    @Column(nullable = false)
    val isGeneral: Boolean,

    @Column(nullable = false)
    val isDress: Boolean,

    @Column(nullable = false)
    val isRehearsalDay: Boolean,

    @Column(nullable = false)
    val isInConcertClothing: Boolean,

    @Column
    @OneToMany(cascade = [CascadeType.ALL], orphanRemoval = true)
    val parts: List<RehearsalPart>
)
